src/main/java/jenkins/plugins/http_request/HttpRequest.java

```diff
@@ -48,56 +54,92 @@
 import org.apache.http.impl.client.SystemDefaultHttpClient;
 import org.apache.http.protocol.BasicHttpContext;
 import org.apache.http.protocol.HttpContext;
+import org.apache.http.HttpEntity;
 import org.apache.http.util.EntityUtils;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
 
 /**
  * @author Janario Oliveira
  */
-public class HttpRequest extends Builder {
-
-    private final String url;
-    private HttpMode httpMode;
-    private MimeType contentType;
-    private MimeType acceptType;
-    private final String outputFile;
-    private final String authentication;
-    private Boolean consoleLogResponseBody;
-    private Boolean passBuildParameters;
-    private List<NameValuePair> customHeaders = new ArrayList<NameValuePair>();
-    private final Integer timeout;
-    private String validResponseCodes;
-    private String validResponseContent;
-
-    /**
-     * @deprecated only to deserialize and serialize in a new form
-     */
-    @Deprecated
-    private Boolean returnCodeBuildRelevant;
+public class HttpRequest extends Builder implements SimpleBuildStep {
+
+    private @Nonnull String url;
+    private HttpMode httpMode                 = DescriptorImpl.defaultHttpMode;
+    private Boolean passBuildParameters       = DescriptorImpl.defaultPassBuildParameters;
+    private String validResponseCodes         = DescriptorImpl.defaultValidResponseCodes;
+    private String validResponseContent       = DescriptorImpl.defaultValidResponseContent;
+    private MimeType acceptType               = DescriptorImpl.defaultAcceptType;
+    private MimeType contentType              = DescriptorImpl.defaultContentType;
+    private String outputFile                 = DescriptorImpl.defaultOutputFile;
+    private Integer timeout                   = DescriptorImpl.defaultTimeout;
+    private Boolean consoleLogResponseBody    = DescriptorImpl.defaultConsoleLogResponseBody;
+    private String authentication             = DescriptorImpl.defaultAuthentication;
+    private List<NameValuePair> customHeaders = DescriptorImpl.defaultCustomHeaders;
+
+    private TaskListener listener;
 
     @DataBoundConstructor
-    public HttpRequest(String url, HttpMode httpMode, String authentication, MimeType contentType,
-                       MimeType acceptType, String outputFile, Boolean returnCodeBuildRelevant,
-                       Boolean consoleLogResponseBody, Boolean passBuildParameters,
-                       List<NameValuePair> customHeaders, Integer timeout,
-                       String validResponseCodes, String validResponseContent)
-            throws URISyntaxException {
+    public HttpRequest(@Nonnull String url)
```

Comment:
The [workflow compatibility guidelines](https://github.com/jenkinsci/workflow-plugin/blob/master/COMPATIBILITY.md#constructor-vs-setters) state: "It is a good idea to replace a lengthy @DataBoundConstructor with a short one taking just truly mandatory parameters (such as a server location)."

The issue I have found with the long constructor is that the snippet generator generates a very long list of arguments. I would prefer to list only the necessary arguments. I am not sure how to do that while maintaining the long constructor.